adler32.c

```diff
@@ -7,6 +7,10 @@
 
 #include "zutil.h"
 
+#if (defined(__ARM_NEON__) || defined(__ARM_NEON))
+#include "adler32_neon.h"
```

Comment:
Put the function definition here, then there is no need for having a adler32_neon.h at all.
As I mentioned on IRC, this is the way it is done elsewhere in the code. That will also solve the compilation problem.